From e88fd364e16a41ceb8db06bcf1275c9fb34b60dd Mon Sep 17 00:00:00 2001
From: Jiaxun Yang <jiaxun.yang@flygoat.com>
Date: Fri, 28 Oct 2022 21:53:30 +0100
Subject: [PATCH] HACK: cevt-r4k: Don't call event_handler if it's not
 registered

After SMP secondary bringup, there will be a very short peroid that
IP7 is enabled but event_handler is not registered.

Just don't call it when suprious TI interrupt comes.
We should investigate proper STATUS_IM settings after bring-up.

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>
---
 arch/mips/kernel/cevt-r4k.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/mips/kernel/cevt-r4k.c b/arch/mips/kernel/cevt-r4k.c
index 368e8475870f..055c025f6f47 100644
--- a/arch/mips/kernel/cevt-r4k.c
+++ b/arch/mips/kernel/cevt-r4k.c
@@ -151,7 +151,8 @@ irqreturn_t c0_compare_interrupt(int irq, void *dev_id)
 		/* Clear Count/Compare Interrupt */
 		write_c0_compare(read_c0_compare());
 		cd = &per_cpu(mips_clockevent_device, cpu);
-		cd->event_handler(cd);
+        if (cd->event_handler)
+			cd->event_handler(cd);
 
 		return IRQ_HANDLED;
 	}
-- 
2.39.1

